src/mlpack/methods/dbscan/dbscan_main.cpp

```diff
@@ -76,6 +76,8 @@ PARAM_FLAG("single_mode", "If set, single-tree range search (not dual-tree) "
     "will be used.", "S");
 PARAM_FLAG("naive", "If set, brute-force range search (not tree-based) "
     "will be used.", "N");
+PARAM_FLAG("random_selection", "If set, random point selection (not ordered) "
```

Comment:
Should we allow the user to select the selection policy, something like: https://github.com/mlpack/mlpack/blob/376c17e11d2797e258b9c1c34aa393d953eccd22/src/mlpack/methods/kernel_pca/kernel_pca_main.cpp#L112-L113

So if we add another strategy we don't have to update the cmd API?

Reply:
Hi, @zoq.
**Q1) Should we allow the user to select the selection policy**
-> In my opinion, currently, there are only two selection policies in DBSCAN, so it is okay to use a binary flag format, but if a policy is added in the future, it would be better to change the parameter flag format like `kernel_pca_main.cpp`.

**Q2) So if we add another strategy we don't have to update the cmd API?**
-> Because I edited the `dbscan_main.cpp` assuming there are only two possible selection policies, If another selection policy is added in the future, in addition to the parameter flag for user to choose the added selection policy, we should update the `dbscan_main` code.

Reply:
I don't have much of an opinion here either way.  The string option means we don't need to break any reverse compatibility when adding new options, though, which is nice.  Whatever you and @zoq choose to do is fine with me. :+1:

Reply:
I would go with the kernal pca strategy, but if @KimSangYeon-DGU likes the current solution, that fine with me as well.

Reply:
I will try it. :)

Reply:
@zoq I changed the code like `mlpack/src/mlpack/methods/kernel_pca/kernel_pca_main.cpp`.